### PR TITLE
Fix several flaky integration tests

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix several flaky integration tests.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
> failed:2

@alesapin Missing whitespace after colon.

**Comment by alexey-milovidov** [comment]
And we have two more:

```
test_insert_distributed_load_balancing/test.py::test_prefer_localhost_replica_0_load_balancing_nearest_hostname
test_ttl_move/test.py::test_concurrent_alter_with_ttl_move[concurrently_altering_ttl_mt-MergeTree()]
```

**Comment by alesapin** [comment]
> > failed:2
> 
> @alesapin Missing whitespace after colon.

**Comment by alesapin** [comment]
> And we have two more:
> 
> ```
> test_insert_distributed_load_balancing/test.py::test_prefer_localhost_replica_0_load_balancing_nearest_hostname
> test_ttl_move/test.py::test_concurrent_alter_with_ttl_move[concurrently_altering_ttl_mt-MergeTree()]
> ```

lol :)

File: src/Storages/Distributed/DirectoryMonitor.cpp [hunk 1]
```diff
@@ -280,8 +279,11 @@ void StorageDistributedDirectoryMonitor::processFile(const std::string & file_pa
         Settings insert_settings;
         std::string insert_query;
         ClientInfo client_info;
+
         readHeader(in, insert_settings, insert_query, client_info, log);
 
+        auto connection = pool->get(timeouts, &insert_settings);
```
Comment (azat) [inline src/Storages/Distributed/DirectoryMonitor.cpp:285]: Nice! And FWIW `dist_settings` can be removed from here:

https://github.com/ClickHouse/ClickHouse/blob/58786f992988a16a8ea787fe624c791529343702/tests/integration/test_insert_distributed_load_balancing/test.py#L37

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**